[rocBLAS] [hipBLAS] CHANGELOG updates for 7.1 (#1738)
ROCm/rocm-libraries: projects/hipblas/CHANGELOG.md

* `--clients-only` build option to only build clients against a prebuilt library
* gfx1103, gfx1150, gfx1151, gfx1200, and gfx1201 support to clients
* FORTRAN enabled for windows build and testing

> * Addition reference library fallback options added

Reviewer: ```suggestion
* `--clients-only` build option to only build clients against a prebuilt library.
* gfx1103, gfx1150, gfx1151, gfx1200, and gfx1201 support to clients.
* FORTRAN enabled for the Microsoft Windows build and tests.
* Additional reference library fallback options added.
```

Changed

> * Improve build time of clients by removing clients_common.cpp from hipblas-test build

Reviewer: ```suggestion
### Changed

* Improve the build time for clients by removing `clients_common.cpp` from the hipblas-test build.
```
